qualtran/surface_code/data_block.py

```diff
@@ -14,113 +14,170 @@
 
-    A surface code layout is segregated into qubits dedicated to magic state distillation
-    and others dedicated to storing the actual data being processed. The latter area is
-    called the data block, and we provide its costs here.
+    The space occupied by the data block is to be contrasted with the space used for magic
+    state distillation.
     """
 
+    @property
     @abc.abstractmethod
-    def footprint(self, n_algo_qubits: int) -> int:
-        """The number of physical qubits used by the data block.
+    def data_d(self):
+        """The code distance used to store the data in the data block."""
+
+    @abc.abstractmethod
+    def n_tiles(self, n_algo_qubits: int) -> int:
+        """The number of surface code tiles used to store a given number of algorithm qubits.
+
+         We define an “algorithm qubit” to be a qubit used in the routing of algorithm-relevant
+         quantum data in a bloq. A physical qubit is a physical system that can encode one qubit,
+         albeit noisily. Specific to the surface code, we define a “tile” to be the minimal area
+         of physical qubits necessary to encode one logical qubit to a particular code distance.
+         A tile can store an algorithm qubit, can be used for ancillary purposes like routing,
+         or can be left idle. A tile is usually a square grid of $2d^2$ physical qubits.
 
-        Attributes:
-            n_algo_qubits: The number of algorithm qubits whose data must be stored and
-                accessed.
+         DataBlock implementations must override this method. This method is used by
+         `self.n_phys_qubits` to report the total number of physical qubits.
+
+        Args:
+            n_algo_qubits: The number of algorithm qubits to compute the number of tiles for.
+
+        Returns:
+            The number of tiles used by this data block to store the given number of algorithm
+            qubits.
         """
 
+    @property
     @abc.abstractmethod
-    def data_error(self, n_algo_qubits: int, n_cycles: int, phys_err: float) -> float:
-        """The error associated with storing data on `n_algo_qubits` for `n_cycles`."""
+    def n_steps_to_consume_a_magic_state(self):
+        """The number of surface code steps to consume a magic state.
 
-    @abc.abstractmethod
-    def n_cycles_to_consume_a_magic_state(self) -> int:
-        """The worst case number of cycles needed to consume a magic state."""
+        We must teleport in "magic states" to do non-Clifford operations on our algorithmic
+        data qubits. The layout of the data block can limit the number magic states consumed
+        per unit time.
+
+        One surface code step is `data_d` cycles of error correction.
+
+        DataBlock imlpementation must override this method. This method is used by
+        `self.n_cycles` to report the total number of cycles required.
+        """
+
+    def n_cycles(self, n_logi_gates: 'MagicCount', logi_err_model: 'LogicalErrorModel') -> int:
```

Comment:
`logi`? 
![image](https://github.com/user-attachments/assets/e4c12caa-2b2f-40cb-9e95-3a69f20d7733)

Reply:
"logical" :D 

naming things is hard, and -- in this module in particular -- there's a balance between ambiguity and readability. I'd like to use consistent, unambiguous names throughout. That means the variable names could get quite long if we don't abbreviate; so I'd like to abbreviate

 - `num_physical_qubits -> n_phys_qubits`
 - `num_logical_gates -> n_logi_gates`

Also the `MagicCount` type annotation will change to `GateCounts` from `qualtran.resource_counting` which will make the variable name make more sense

Reply:
> naming things is hard, and -- in this module in particular -- there's a balance between ambiguity and readability

you are the native english speaker in this conversation so I will let this to your judgement 

---


> Also the `MagicCount` type annotation will change to `GateCounts`

wouldn't that be confusing? since GateCounts could imply all gates including say cliffords?!

Reply:
this is an abstract interface, so yes -- it should generally be able to depend on the gate counts. If the implementation doesn't care about clifford counts it can ignore them. You can imagine a data block being able to do only a certain number of cliffords per cycle

Reply:
abbreviating `logical` -> `logi` seems very awkward to me as well. I'd have a preference towards using IDE to manage the longer names but avoid non-standard abbreviations, especially for `logical` -> `logi`. 

Abbreviating by deleting letters of a word is also generally [discouraged by the google style guide](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules). Please consider using the full word for `logical`

Reply:
I mean "phys" and "prob" are already used everywhere

Reply:
I've updated all the things within the scope of this PR to use full names